zerver/views/registration.py

```diff
@@ -363,28 +378,84 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
-                 # we have to set it here.
-                 'creating_new_team': realm_creation,
-                 'password_required': password_auth_enabled(realm) and password_required,
-                 'require_ldap_password': require_ldap_password,
-                 'password_auth_enabled': password_auth_enabled(realm),
-                 'root_domain_available': is_root_domain_available(),
-                 'default_stream_groups': get_default_stream_groups(realm),
-                 'accounts': get_accounts_for_email(email),
-                 'MAX_REALM_NAME_LENGTH': str(Realm.MAX_REALM_NAME_LENGTH),
-                 'MAX_NAME_LENGTH': str(UserProfile.MAX_NAME_LENGTH),
-                 'MAX_PASSWORD_LENGTH': str(form.MAX_PASSWORD_LENGTH),
-                 'MAX_REALM_SUBDOMAIN_LENGTH': str(Realm.MAX_REALM_SUBDOMAIN_LENGTH)
-                 }
+        context = {
+            'form': form,
+            'email': email,
+            'key': key,
+            'full_name': request.session.get('authenticated_full_name', None),
+            'lock_name': name_validated and name_changes_disabled(realm),
+            # password_auth_enabled is normally set via our context processor,
+            # but for the registration form, there is no logged in user yet, so
+            # we have to set it here.
+            'creating_new_team': realm_creation,
+            'billing_enabled': settings.BILLING_ENABLED,
+            'password_required': password_auth_enabled(realm) and password_required,
+            'require_ldap_password': require_ldap_password,
+            'password_auth_enabled': password_auth_enabled(realm),
+            'root_domain_available': is_root_domain_available(),
+            'default_stream_groups': get_default_stream_groups(realm),
+            'accounts': get_accounts_for_email(email),
+            'MAX_REALM_NAME_LENGTH': str(Realm.MAX_REALM_NAME_LENGTH),
+            'MAX_NAME_LENGTH': str(UserProfile.MAX_NAME_LENGTH),
+            'MAX_PASSWORD_LENGTH': str(form.MAX_PASSWORD_LENGTH),
+            'MAX_REALM_SUBDOMAIN_LENGTH': str(Realm.MAX_REALM_SUBDOMAIN_LENGTH)
+        }
     )
 
+def notify_support_new_realm(realm: Realm, organization_type: str,
+                             repository_link: str,
+                             organization_nature: str,
+                             organization_website: str,
+                             organization_description: str) -> None:
+
+    context = common_context_unauthed(realm)
+
+    if settings.STAFF_SUBDOMAIN is not None:
+        try:
+            realm_uri = get_realm(settings.STAFF_SUBDOMAIN).uri + reverse('analytics.views.support')
+            staff_realm_uri = realm_uri + "/?q=" + realm.string_id
+            context.update({"realm_uri": staff_realm_uri})
+        except Realm.DoesNotExist:
+            return HttpResponseNotFound("Realm %s does not exist" % (settings.STAFFSUBDOMAIN,))
```

Comment:
I think we wanted to use `realm.uri` in this case, when the STAFF_SUBDOMAIN thing fails? :thinking:

Reply:
It is actually giving `realm.uri` when STAFF_DOMAIN is not set. The `context` initially have `realm.uri` value which is for the realm. The value of `realm.uri` only changes to `staff_realm_uri` when settings.STAFF_SUBDOMAIN is present. If it is not, then it will not enter the if statement and the value of `realm.uri` remains the same.

Reply:
Ah, I see! I think a good comment explaining that would be good to include in the code. And a typo in `"Realm %s does not exist" % (settings.STAFFSUBDOMAIN,)`